packages/liveblocks-react/src/comments/CommentsRoom.ts

```diff
@@ -93,166 +118,262 @@ export type RoomThreads<TThreadMetadata extends BaseMetadata> =
     }
   | {
       isLoading: false;
-      threads: ThreadData<TThreadMetadata>[];
+      threads: Data;
       error?: never;
     };
 
-export function createCommentsRoom<TThreadMetadata extends BaseMetadata>(
-  room: Room<JsonObject, LsonObject, BaseUserMeta, Json>,
-  errorEventSource: EventSource<CommentsApiError<TThreadMetadata>>
-): CommentsRoom<TThreadMetadata> {
-  const store = createStore<RoomThreads<TThreadMetadata>>({
-    isLoading: true,
-  });
-
-  let fetchThreadsPromise: Promise<ThreadData<TThreadMetadata>[]> | null = null;
-
-  // Temporary solution
-  // The most basic conflict resolution
-  // If there are any pending mutation, we simply ignore any threads coming from the server
-  // When all mutations are finished, we pull the source of truth from the backend
-
-  let numberOfMutations = 0;
-  function endMutation() {
-    numberOfMutations--;
-    if (numberOfMutations === 0) {
-      void revalidateThreads();
-    }
-  }
+type RequestInfo<Data = any> = {
+  fetcher: Promise<Data>;
+  timestamp: number;
+};
+
+type MutationInfo = {
+  startTime: number;
+  endTime: number;
+};
+
+class StateManager<Data = any> extends EventTarget {
+  private _cache: State<Data> | undefined; // Stores the current cache state (threads)
```

Comment:
Chose to use the `EventTarget` Web API for the `StateManager` class as it aligns well with its browser-only usage context and doesn't require the additional methods provided by the `makeEventSource` API. However, I remain open to switching to `makeEventSource` if there is a stronger team preference for it.

Reply:
I don't have strong preferences on these, maybe @nvie?

Reply:
The code already ships, and it's used everywhere else, so I'd personally prefer that. Practically speaking, in most usage contexts, it's easier to use too because it returns the unsubscribe function directly, so where you use it, you don't have to go through the ceremony of naming the subscribe function and then making sure to pass the exact same function reference to the unsubscribe function later. And lastly, using the event source API ensures type-safety nicely. It's impossible to call `eventSource.notify(...)` with a param that isn't of the expected type, unlike doing `.dispatchEvent(...)`, which you can pass any event (type). So IMO, code maintenance is much easier with the event source API — it's why it exists.

Reply:
Just added a comment related to this: https://github.com/liveblocks/liveblocks/pull/1142#discussion_r1333316264

> Updated to use `EventSource` API. One of the reasons why I opted to use `EventTarget` Web API was that I wanted to use standard Web APIs, so anyone reading the code in future won't have to learn a new API if they already knew the `EventTarget` API but I have to agree that this `EventSource` API provides a much better experience. Thanks for the suggestion @nvie